src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -592,6 +592,14 @@ private function addService($id, $definition)
             $return[] = sprintf("@throws InactiveScopeException when the '%s' service is requested while the '%s' scope is not active", $id, $scope);
         }
 
+        if ($definition->isDeprecated()) {
+            if ($return && 0 === strpos($return[count($return) - 1], '@return')) {
+                $return[] = '';
+            }
+
+            $return[] = '@deprecated';
```

Comment:
Having a recommandation does **not** erase the fact that the default message is not useful enough to be put here. Having a recommandation is saying "you **should**, not you **must**....

I could then make the default message optionnal AND if it is null, use the default. Then in the `@deprecated` part, if the "message" is *not* null, then add it here... But doesn't that honestly seem overkill ?